packages/neuron-wallet/src/models/dao-utils.ts

```diff
@@ -0,0 +1,52 @@
+import NodeService from 'services/node'
+import { ScriptHashType } from 'types/cell-types'
+import Core from '@nervosnetwork/ckb-sdk-core'
+import { SystemScript } from './lock-utils'
+import { scriptToHash } from '@nervosnetwork/ckb-sdk-utils'
+
+export default class DaoUtils {
+  daoScript: SystemScript
+
+  constructor(daoScript: SystemScript) {
+    this.daoScript = daoScript
+  }
+
+  static daoScriptInfo: SystemScript | undefined
+
+  static previousURL: string | undefined
+
+  static async loadDaoScript(nodeURL: string): Promise<SystemScript> {
+    const core = new Core(nodeURL)
+
+    const genesisBlock = await core.rpc.getBlockByNumber(BigInt(0))
+    const systemCellTransaction = genesisBlock.transactions[0]
+    const daoOutPoint = {
+      txHash: systemCellTransaction.hash,
+      index: '0x2'
+    }
+
+    const daoTypeHash = scriptToHash(systemCellTransaction.outputs[2].type!)
+
+    return {
+      codeHash: daoTypeHash,
+      outPoint: daoOutPoint,
+      hashType: ScriptHashType.Type
+    }
+  }
+
+  static async daoScript(nodeURL: string = NodeService.getInstance().core.rpc.node.url): Promise<SystemScript> {
```

Comment:
Is it on purpose to enable `DaoUtils.daoScript` and `DaoUtilsIns.daoScript` ?

Reply:
What you means ?

Reply:
https://github.com/nervosnetwork/neuron/pull/1050/files/5c6cc26dcc1b03aecf350f52eb07bdf715c036ee#diff-355c492b084406998c9e61f77cd48d72R8 defines `DaoUtilsIns.daoScript`

https://github.com/nervosnetwork/neuron/pull/1050/files/5c6cc26dcc1b03aecf350f52eb07bdf715c036ee#diff-355c492b084406998c9e61f77cd48d72R37 defines `DaoUtils.daoScript`

Reply:
Oh I see, this can refer to `LockUtils` 
can use it like this, for better to get `daoScript` for instance methods, not `async/await`
`new LockUtils(await LockUtils.systemScript()).addressesToAllLockHashes(addresses)`
It not used now.